src/utils/blogs.ts

```diff
@@ -91,21 +98,34 @@ export async function getBlogBySlug<F extends (keyof Blog)[]>(
 
   const date = typeof data.date === 'string' ? data.date : fs.statSync(fullPath).birthtime.toISOString()
   const readingTime = Math.round(content.length / 1300).toString()
-
-  const authorNames: string[] = Array.isArray(data.author)
-    ? data.author.filter((author): author is string => typeof author === 'string')
-    : typeof data.author === 'string'
-    ? [data.author]
-    : ['Nervos']
-
-  const authors = authorNames.map(name => {
-    if (name.toLowerCase() === 'nervos') return { name, avatar: '/images/nervos_avatar.svg' }
-    if (name.startsWith('github:')) {
-      name = name.substring('github:'.length)
-      return { name, avatar: `https://avatars.githubusercontent.com/${name}` }
+  const authorStringFilter = (author: unknown): author is string => typeof author === 'string'
+  const authorWithDetailFilter = (author: unknown): author is AuthorDetail =>
+    Object.prototype.toString.call(author) === '[object Object]' &&
+    typeof (author as AuthorDetail).name === 'string' &&
+    typeof (author as AuthorDetail).url === 'string' &&
+    (author as AuthorDetail).platform === 'github'
+
+  const authorArray = Array.isArray(data.author) ? data.author : [data.author]
+  const stringAuthors = authorArray.filter(authorStringFilter)
+  const objectAuthors = authorArray.filter(authorWithDetailFilter)
+  const filteredAuthors = [...stringAuthors, ...objectAuthors]
```

Comment:
It's a bit complicated and reorders the list.

How about

```js
const list = Array.isArray(data.author) ? data.author : [data.author]
const authorList = list.map(author => {
  if (typeof author === 'string') {
    return { name: author }
  }
  return author
})
.filter(author => author.name)
```

Reply:
`return { name: author }` changes the data structure, I just simplified the logic by [2f7d99d](https://github.com/Magickbase/nervos-official-website/pull/339/commits/2f7d99df1c890239e0845da7b4b9fb39b6c22a57)

Reply:
This explaination is straightforward, updated in [ca8f8fd](https://github.com/Magickbase/nervos-official-website/pull/339/commits/ca8f8fd380a9c8bd24c9af8892a169fd57b31c88)